xhype/xhype/src/vmexit.rs

```diff
@@ -13,6 +16,105 @@ pub enum HandleResult {
     Next,
 }
 
+// Table 27-3
+#[inline]
+pub fn vmx_guest_reg(num: u64) -> X86Reg {
+    match num {
+        0 => X86Reg::RAX,
+        1 => X86Reg::RCX,
+        2 => X86Reg::RDX,
+        3 => X86Reg::RBX,
+        4 => X86Reg::RSP,
+        5 => X86Reg::RBP,
+        6 => X86Reg::RSI,
+        7 => X86Reg::RDI,
+        8 => X86Reg::R8,
+        9 => X86Reg::R9,
+        10 => X86Reg::R10,
+        11 => X86Reg::R11,
+        12 => X86Reg::R12,
+        13 => X86Reg::R13,
+        14 => X86Reg::R14,
+        15 => X86Reg::R15,
+        _ => panic!("bad register in exit qualification"),
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// VMX_REASON_MOV_CR
+////////////////////////////////////////////////////////////////////////////////
+
+// Intel SDM Table 27-3.
+pub fn handle_cr(vcpu: &VCPU, _gth: &GuestThread) -> Result<HandleResult, Error> {
+    let qual = vcpu.read_vmcs(VMCS_RO_EXIT_QUALIFIC)?;
+    let ctrl_reg = match qual & 0xf {
+        0 => X86Reg::CR0,
+        3 => X86Reg::CR3,
+        4 => X86Reg::CR4,
+        8 => {
+            return Err(Error::Unhandled(
+                VMX_REASON_MOV_CR,
+                format!("access cr8: unimplemented"),
```

Comment:
What `format!` here? Can you not pass a `&'static str` here?

Reply:
The definition in `xhype::err::Error` is `Unhandled(u64, String)`, so I need to pass a `String`. I use `format!` to do the conversion. For sure I can also use `.to_string()`.

Reply:
I see. Yeah, `.to_string()` or even `.into()` would be preferable here.